airbyte-scheduler/client/src/main/java/io/airbyte/scheduler/client/DefaultSynchronousSchedulerClient.java

```diff
@@ -102,25 +108,32 @@ public SynchronousResponse<ConnectorSpecification> createGetSpecJob(final String
         ConfigType.GET_SPEC,
         null,
         jobId -> temporalClient.submitGetSpec(UUID.randomUUID(), 0, jobSpecConfig),
+        ConnectorJobOutput::getSpec,
         null);
   }
 
   @VisibleForTesting
-  <T> SynchronousResponse<T> execute(final ConfigType configType,
-                                     @Nullable final UUID connectorDefinitionId,
-                                     final Function<UUID, TemporalResponse<T>> executor,
-                                     final UUID workspaceId) {
+  <T, U> SynchronousResponse<T> execute(final ConfigType configType,
+                                        @Nullable final UUID connectorDefinitionId,
+                                        final Function<UUID, TemporalResponse<U>> executor,
+                                        final Function<U, T> outputMapper,
+                                        final UUID workspaceId) {
     final long createdAt = Instant.now().toEpochMilli();
     final UUID jobId = UUID.randomUUID();
     try {
       track(jobId, configType, connectorDefinitionId, workspaceId, JobState.STARTED, null);
-      final TemporalResponse<T> operationOutput = executor.apply(jobId);
-      final JobState outputState = operationOutput.getMetadata().isSucceeded() ? JobState.SUCCEEDED : JobState.FAILED;
-      track(jobId, configType, connectorDefinitionId, workspaceId, outputState, operationOutput.getOutput().orElse(null));
-      final long endedAt = Instant.now().toEpochMilli();
+      final TemporalResponse<U> temporalResponse = executor.apply(jobId);
+      final Optional<U> jobOutput = temporalResponse.getOutput();
+      final T mappedOutput = jobOutput.map(outputMapper).orElse(null);
+      final JobState outputState = temporalResponse.getMetadata().isSucceeded() ? JobState.SUCCEEDED : JobState.FAILED;
 
+      track(jobId, configType, connectorDefinitionId, workspaceId, outputState, mappedOutput);
+      // TODO(pedro): report ConnectorJobOutput's failureReason to the JobErrorReporter, like the above
```

Comment:
To keep this PR somewhat manageable I've limited it to only building/surfacing the failure reasons for these jobs, but not actually reporting it to the JobErrorReporter (sentry). This is where that would happen, following the same pattern that's being used for the JobTracker (segment). This will come in a subsequent PR.